mlflow/metrics/utils/prompt_template.py

```diff
@@ -0,0 +1,50 @@
+import re
+
+
+class PromptTemplate:
+    """A prompt template for a language model.
+
+    A prompt template consists of a string template. It accepts a set of parameters
+    from the user that can be used to generate a prompt for a language model.
+
+    The template can be formatted using f-strings.
+
+    Example:
+
+        .. code-block:: python
+
+            from mlflow.metrics.utils.prompt_template import PromptTemplate
+
+            # Instantiation using initializer
+            prompt = PromptTemplate(template_str="Say {foo} {baz}", variables=["foo", "baz"])
+
+            # Instantiation using partial_fill
+            prompt = PromptTemplate(template_str="Say {foo} {baz}").partial_fill(foo="bar")
```

Comment:
```suggestion
            prompt = PromptTemplate(template_str="Say {foo} {baz}").partial_fill(foo="bar")
            assert prompt == "Say bar {baz}"
```

to clarify what `partial_fill` returns.